pyk/src/pyk/rpc/rpc.py

```diff
@@ -71,37 +72,105 @@ class JsonRpcMethod(Protocol):
     def __call__(self, **kwargs: Any) -> Any: ...
 
 
-class JsonRpcRequestHandler(BaseHTTPRequestHandler):
-    methods: dict[str, JsonRpcMethod]
+@dataclass(frozen=True)
+class JsonRpcRequest:
 
-    def __init__(self, methods: dict[str, JsonRpcMethod], *args: Any, **kwargs: Any) -> None:
-        self.methods = methods
-        super().__init__(*args, **kwargs)
+    method: str
+    params: Any
+    id: Any
+
+    @staticmethod
+    def validate(request_dict: Any, valid_methods: Iterable[str]) -> JsonRpcRequest | JsonRpcError:
```

Comment:
Consider moving this to `JsonRpcRequestHandler`.

Reply:
Then `valid_methods` can also be removed from the parameters (assuming the method is not static).